src/routes/sign/Sign.svelte

```diff
@@ -1,7 +1,56 @@
 <script lang="ts">
+    import { sendMessage } from '../../helpers/sendMessageHelpers';
     import Transaction from '../../lib/transaction/Transaction.svelte';
+    import { getParamsFromUrl } from './signHelpers';
+    import { xdr, description, isXdrNull } from './signStore';
+    import type IxdrInvalid from '../../lib/errors/IxdrInvalid';
+    import type { IReadyEvent } from 'src/helpers/eventInterfaces/IReadyEvent';
+
+    let readyEvent: IReadyEvent = {
+        type: 'ready',
+        message: 'Simple Signer is ready to operate',
+    };
+
+    function messageHandler(e: MessageEvent) {
+        if ('xdr' in e.data && 'description' in e.data) {
+            $xdr = e.data.xdr;
+            $description = e.data.description;
+        } else if ('xdr' in e.data) {
+            $xdr = e.data.xdr;
+        }
```

Comment:
split this in two separate if statements

if xdr....
  set xdr
else
  fail

if description...
  set descripcion